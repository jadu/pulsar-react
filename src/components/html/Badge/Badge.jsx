import React from 'react';
import classnames from 'classnames';

export default class Button extends React.Component {

  static defaultProps = {
    className: 'badge'
  };

  render() {
    let {
      children,
      className,
      danger,
      info,
      inverse,
      naked,
      outline,
      primary,
      small,
      success,
      warning,
      white,
      ...props
    } = this.props;

    let variantClasses = classnames(className, {
      'badge--danger': danger,
      'badge--info': info,
      'badge--inverse': inverse,
      'badge--primary': primary,
      'badge--success': success,
      'badge--warning': warning,
      'badge--white': white
    });

    return (
      <React.Fragment>
        <span 
          className={variantClasses}
          {...props}
        >
          {children}
        </span>
      </React.Fragment>
    );
  }
}
