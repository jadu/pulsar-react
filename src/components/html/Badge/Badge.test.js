'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import Badge from './Badge';

it('renders the base badge', () => {
  const tree = render(
    <Badge>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the primary variation', () => {
  const tree = render(
    <Badge primary>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the info variation', () => {
  const tree = render(
    <Badge info>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the success variation', () => {
  const tree = render(
    <Badge success>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the warning variation', () => {
  const tree = render(
    <Badge warning>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the danger variation', () => {
  const tree = render(
    <Badge danger>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the inverse variation', () => {
  const tree = render(
    <Badge inverse>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the white variation', () => {
  const tree = render(
    <Badge white>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the outline variation', () => {
  const tree = render(
    <Badge outline>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the naked variation', () => {
  const tree = render(
    <Badge naked>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders the small variation', () => {
  const tree = render(
    <Badge small>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders multiple variations', () => {
  const tree = render(
    <Badge primary outline small>foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});

it('renders as a link if the href attribute is supplied', () => {
  const tree = render(
    <Badge href="#bar">foo</Badge>
  );
  expect(tree).toMatchSnapshot();
});