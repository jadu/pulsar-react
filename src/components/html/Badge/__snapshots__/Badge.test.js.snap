// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as a link if the href attribute is supplied 1`] = `
<span
  class="badge"
  href="#bar"
>
  foo
</span>
`;

exports[`renders multiple variations 1`] = `
<span
  class="badge badge--primary"
>
  foo
</span>
`;

exports[`renders the base badge 1`] = `
<span
  class="badge"
>
  foo
</span>
`;

exports[`renders the danger variation 1`] = `
<span
  class="badge badge--danger"
>
  foo
</span>
`;

exports[`renders the info variation 1`] = `
<span
  class="badge badge--info"
>
  foo
</span>
`;

exports[`renders the inverse variation 1`] = `
<span
  class="badge badge--inverse"
>
  foo
</span>
`;

exports[`renders the naked variation 1`] = `
<span
  class="badge"
>
  foo
</span>
`;

exports[`renders the outline variation 1`] = `
<span
  class="badge"
>
  foo
</span>
`;

exports[`renders the primary variation 1`] = `
<span
  class="badge badge--primary"
>
  foo
</span>
`;

exports[`renders the small variation 1`] = `
<span
  class="badge"
>
  foo
</span>
`;

exports[`renders the success variation 1`] = `
<span
  class="badge badge--success"
>
  foo
</span>
`;

exports[`renders the warning variation 1`] = `
<span
  class="badge badge--warning"
>
  foo
</span>
`;

exports[`renders the white variation 1`] = `
<span
  class="badge badge--white"
>
  foo
</span>
`;
